qcengine/programs/adcc.py

```diff
@@ -0,0 +1,159 @@
+"""
+Calls adcc
+"""
+import json
+import os
+import sys
+from pathlib import Path
+import importlib
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple
+import numpy as np
+
+from qcelemental.models import AtomicResult, AtomicResultProperties, Provenance
+from qcelemental.util import deserialize, parse_version, safe_version, which_import
+
+from ..exceptions import InputError, RandomError, ResourceError, UnknownError
+from .model import ProgramHarness
+
+if TYPE_CHECKING:
+    from qcelemental.models import AtomicInput
+
+    from ..config import TaskConfig
+
+
+class AdccHarness(ProgramHarness):
+
+    _defaults = {
+        "name": "adcc",
+        "scratch": False,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    @staticmethod
+    def found(raise_error: bool = False) -> bool:
+        """Whether adcc harness is ready for operation.
+        Parameters
+        ----------
+        raise_error: bool
+            Passed on to control negative return between False and ModuleNotFoundError raised.
+        Returns
+        -------
+        bool
+            If adcc and psi4 are found, returns True.
+            If raise_error is False and adcc or psi4 is missing, returns False.
+            If raise_error is True and adcc or psi4 are missing, the error message is raised.
+        """
+        found_adcc = which_import(
+            "adcc",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install via `conda install adcc -c adcc`.",
+        )
+        found_psi4 = which_import(
+            "psi4",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install psi4 for adcc harness via `conda install psi4 -c psi4`.",
+        )
+        return found_adcc and found_psi4
+
+    def get_version(self) -> str:
+        """Return the currently used version of adcc"""
+        self.found(raise_error=True)
+
+        which_prog = which_import("adcc")
+        if which_prog not in self.version_cache:
+            import adcc
+
+            self.version_cache[which_prog] = safe_version(adcc.__version__)
+        return self.version_cache[which_prog]
+
+    def compute(self, input_model: "AtomicInput", config: "TaskConfig") -> "AtomicResult":
+        """
+        Runs adcc
+        """
+        self.found(raise_error=True)
+        import adcc
+        import psi4
+
+        mol = input_model.molecule
+        model = input_model.model
+        kws = input_model.keywords
+
+        # TODO: smooth way to do this?
+        psi4_molecule = psi4.geometry(mol.to_string(dtype="psi4") + "symmetry c1")
+        psi4.core.clean()
+        psi4.core.be_quiet()
+        psi4.set_options(
```

Comment:
Just a note, because it came to my mind: We should set the psi4 SCF tolerance in line with the adcc tolerance (i.e. to ensure that https://github.com/adc-connect/adcc/blob/55436b4bbf6c8a716c99add3f15adde341acb9ff/adcc/workflow.py#L338 is not triggered ;)

Reply:
done.